packages/charts/src/chart_types/xy_chart/domains/x_domain.ts

```diff
@@ -127,18 +133,77 @@ function getMinInterval(computedMinInterval: number, size: number, customMinInte
   return customMinInterval;
 }
 
+const approximatedOptions: {
+  unit: moment.unitOfTime.Base;
+  min?: number;
+  max?: number;
+}[] = [
+  { unit: 'year' },
+  { unit: 'day', min: 6, max: 8 },
+  { unit: 'week', max: 52 },
+  { unit: 'day', max: 32 },
+  { unit: 'hour', max: 24 },
+  { unit: 'minute', max: 60 },
+  { unit: 'second', max: 60 },
+  { unit: 'millisecond', max: 1000 },
+];
+
+/**
+ * Finds the nearest rounded unit of time that falls between a defined min and max
+ * iteratively starting from the largest unit of time going to the smallest
+ *
+ * TODO: build in rules to apply multiplier of the unit interval (e.g. 12h)
+ */
+function approximateNearestDuration(value: number): number {
+  if (!value) return value;
+
+  for (let i = 0; i < approximatedOptions.length; i++) {
+    const { unit, min = 1, max = 10 } = approximatedOptions[i];
+    const unitInterval = moment.duration(1, unit).asMilliseconds();
+    const unitDiff = Math.floor(value / unitInterval);
+
+    if ((unitDiff > min && unitDiff <= max) || (unit === 'year' && unitDiff > max)) {
+      return unitInterval;
+    }
+  }
+
+  return Math.round(value);
+}
+
+/**
+ * Guess minInterval when only single value exists
+ * @internal
+ */
+export function guessMinInterval([start, end]: number[], scaleType: ScaleContinuousType): number {
+  const domainGap = Math.abs(start - end);
+  const value =
+    scaleType === ScaleType.Time
+      ? approximateNearestDuration(domainGap)
+      : domainGap === 0
+      ? 1
+      : Math.round(domainGap / DEFAULT_INTERVAL_COUNT);
+
+  Logger.warn(
+    `The minInterval, given only a single data point, has been approximated as ${
+      scaleType === ScaleType.Time ? moment.duration(value).humanize() : value
+    }. This is not ideal as this data point likely does not coincide with the proper interval. Please set the minInterval in the xDomain.`,
+  );
+
+  return value;
+}
+
 /**
  * Find the minimum interval between xValues.
- * Default to 0 if an empty array, 1 if one item array
+ * Default to 0 if an empty array, null if one item array
  * @internal
  */
-export function findMinInterval(xValues: number[]): number {
+export function findMinInterval(xValues: number[]): number | null {
   const valuesLength = xValues.length;
   if (valuesLength <= 0) {
     return 0;
   }
   if (valuesLength === 1) {
-    return 1;
+    return null;
```

Comment:
Instead of returning null, can we just call `guessMinInterval(seriesXComputedDomains, fallbackScale ?? type);` here?
```suggestion
    return guessMinInterval(seriesXComputedDomains, fallbackScale ?? type);
```

Reply:
I thought about that but then I would need to add two more arguments to this function. Which would you prefer?

Reply:
fwiw I tend to trade argument list brevity for return type simplicity so I wouldn't mind having two extra args, although there may be some better solution (`NaN` would make it `number` consistently, but if you need an `isNaN` downstream due to this, then it'd be a bit superficial)

Reply:
Putting aside the question around `null` the function is basically a reduction:
```
export function findMinInterval(xValues: number[]): number | null {
  const sortedValues = xValues.slice().sort(compareByValueAsc);
  return xValues.length <= 0
    ? 0
    : xValues.length === 1
    ? null
    : sortedValues.reduce(
        (p, n, i, a) => Math.min(p, (i < a.length - 1 ? a[i + 1] : Infinity) - n),
        sortedValues[1] - sortedValues[0],
      );
}
```
Somehow it feels asymmetrical that for a zero length array, we return zero, and for a 1-length array, it's null, then from 2, it's a number again

Reply:
Oh my point about this being a reduction: probably it belongs in some basic array processing util, not `x_domain.ts`, so 
```
export function findMinInterval(xValues: number[]): number | null {
  const sortedValues = xValues.slice().sort(compareByValueAsc);
  return xValues.length <= 0
    ? 0
    : xValues.length === 1
    ? null
    : minDelta(sortedValues)
}

```